### PR TITLE
Deprecate ServiceHttpClient

### PR DESCRIPTION
**Pull request by jochenschneider**

### CODE REVIEW

File: libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala [hunk 1]
```diff
@@ -18,15 +18,15 @@ import cats.syntax.option._
 import scala.concurrent.Future
 import scala.reflect.ClassTag
 
-
+@deprecated("use data type from libats-http", "v0.4.0-40")
 trait ServiceHttpClientSupport {
```
Comment (simao) [inline libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: Message could be better, this is not a data type

File: libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala [hunk 2]
```diff
@@ -18,15 +18,15 @@ import cats.syntax.option._
...
   def defaultHttpClient(implicit system: ActorSystem, mat: Materializer): (HttpRequest => Future[HttpResponse]) = {
     val _http = Http()
     req => _http.singleRequest(req)
   }
 }
 
-// TODO: Move to libats?
+@deprecated("use data type from libats-http", "v0.4.0-40")
 abstract class ServiceHttpClient(httpClient: HttpRequest => Future[HttpResponse])
```
Comment (simao) [inline libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: Why don't you use the client from `libats-http v0.4.0-40` ?

Comment (jochenschneider) [reply libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: Oops.

File: libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala (hunk 1 above)
Comment (jochenschneider) [reply libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: Ah this is actually for the other PR, right? I am not sure what you mean. A Http client should go into libats-http on the face of it, message deals with Kafka, no?

Comment (simao) [reply libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: yeah the http client should go in libats-http, but I meant the deprecated message, it says "data type" but `ServiceHttpClientSupport` is not a data type.

But anyway this should just go and use the one from libats-http.

Comment (jochenschneider) [reply libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/http/ServiceHttpClient.scala]: Ah, gotcha. I had assumed we'd keep old stuff as deprecated for a while, but I guess it's not really important here.

**Pull request merged**